### PR TITLE
Update google storage library

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit updates cloud.google.com/go/storage to 1.35.

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Upgrade `cloud.google.com/go/storage version` `v1.35.1`

## Verification

Unit tests

### CODE REVIEW

**Review by saswatamcode — APPROVED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -88,8 +88,7 @@ type BucketReader interface {
 	// IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
 	IsObjNotFoundErr(err error) bool
 
-	// IsAccessDeniedErr returns true if access to object is denied.
-	IsAccessDeniedErr(err error) bool
+	// IsAccessDeniedErr returns true if acces to object is denied.
```
Comment (GiedriusS) [inline README.md:91]: This seems a bit off - do we need to edit the mdox line above?

Comment (fpetkovski) [reply README.md]: You're right, the method comment had a typo.

**Comment by fpetkovski** [comment]
@GiedriusS any other comments or are we good to merge this?

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**